spec.html

```diff
@@ -7652,19 +7650,44 @@ <h1>Jobs and Host Operations to Enqueue Jobs</h1>
+    </emu-note>
+
+    <p>Particular kinds of Jobs have additional conformance requirements.</p>
+
     <emu-clause id="sec-hostenqueuepromisejob" aoid="HostEnqueuePromiseJob">
       <h1>HostEnqueuePromiseJob ( _job_, _realm_ )</h1>
       <p>HostEnqueuePromiseJob is a host-defined abstract operation that schedules the Job Abstract Closure _job_ to be performed, at some future time. The Abstract Closures used with this algorithm are intended to be related to the handling of Promises, or otherwise, to be scheduled with equal priority to Promise handling operations.</p>
-      <p>The _realm_ parameter is passed through to hosts with no normative requirements; it is either *null* or a Realm.</p>
+      <p>The _realm_ parameter is either *null* or a Realm Record.</p>
+
+      <p>The implementation of HostEnqueuePromiseJob must conform to the requirements in <emu-xref href="#sec-jobs"></emu-xref>. Additionally, Jobs enqueued by this hook must conform to all of the following requirements:</p>
+      <ul>
+        <li>If _realm_ is not *null*, the implementation must perform implementation-defined steps such that all invocations of _job_ are prepared to evaluate code.</li>
+        <li>Let _scriptOrModule_ be GetActiveScriptOrModule(). The implementation must perform implementation-defined steps such that all invocations of _job_ propagate the script or module _scriptOrModule_.</li>
+        <li>Jobs must be scheduled in FIFO order, with Jobs running in the same order as the HostEnqueuePromiseJob invocations which scheduled them.</li>
+      </ul>
 
       <emu-note>
-        The _realm_ for PromiseResolveThenableJobs is the result of calling GetFunctionRealm on the _then_ function object. The _realm_ for PromiseReactionJobs is the result of calling GetFunctionRealm on the handler if the handler is not *undefined*. Otherwise the _realm_ is *null*. The WHATWG HTML specification, for example, uses _realm_ to check for ability to run script and to prepare to run script.
+        <p>The _realm_ for Jobs returned by NewPromiseResolveThenableJob is usually the result of calling GetFunctionRealm on the _then_ function object. The _realm_ for Jobs returned by NewPromiseReactionJob is usually the result of calling GetFunctionRealm on the handler if the handler is not *undefined*. When the _realm_ is *null* due to being called on a revoked Proxy, the current Realm at the time of the GetFunctionRealm call is used instead. When the _realm_ is *null*, no user ECMAScript code will be evaluated and no new ECMAScript objects (e.g. Error objects) will be created. The WHATWG HTML specification, for example, uses _realm_ to check for ability to run script and for the <a href="https://html.spec.whatwg.org/#entry">entry</a> concept.</p>
```

Comment:
> When the _realm_ is *null* due to being called on a revoked Proxy, the current Realm at the time of the GetFunctionRealm call is used instead.

This sentence is confusing to me. I _think_ it's describing the fact that in `NewPromiseReactionJob` if `GetFunctionRealm` is called on a revoked proxy then the value used for the _realm_ parameter when scheduling the job is the current Realm at the time of the `GetFunctionRealm` call.

But then I don't know why it's talking about "when the `_realm_` is *null*", since I don't know what "the current Realm at the time of the GetFunctionRealm call is used" _for_, if not the `_realm_` parameter, in which case it is not *null* in the first place. 

Should this perhaps say "when `GetFunctionRealm` returns an abrupt completion due to [...]" instead of "when the `_realm_` is null due to [...]"?

Reply:
The part that talks about the realm being null for a RevokedProxy is just wrong. That was originally incorrect and was fixed in the last commit, and I didn't update the note. Will fix.